superset/assets/visualizations/Legend.jsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import './Legend.css';
+
+const propTypes = {
+  categories: PropTypes.object,
+  toggleCategory: PropTypes.func,
+};
+
+const defaultProps = {
+  categories: {},
+  toggleCategory: () => {},
+};
+
+export default class Legend extends React.PureComponent {
```

Comment:
Can we name this `MapLegend` or do you seeing this being extended to all visualizations?

Reply:
My idea was to use this as a legend for all visualization types that don't have a native one (like nvd3 has). Right now this component takes only a list of categories and is not specific to maps.